packages/flutter/lib/src/material/tooltip_theme.dart

```diff
@@ -0,0 +1,227 @@
+      height,
+      padding,
+      verticalOffset,
+      preferBelow,
+      excludeFromSemantics,
+      decoration,
+      textStyle,
+      waitDuration,
+      showDuration,
+    );
+  }
+
+  @override
+  bool operator==(Object other) {
+    if (identical(this, other))
+      return true;
+    if (other.runtimeType != runtimeType)
+      return false;
+    final TooltipThemeData typedOther = other;
+    return typedOther.height == height
+        && typedOther.padding == padding
+        && typedOther.verticalOffset == verticalOffset
+        && typedOther.preferBelow == preferBelow
+        && typedOther.excludeFromSemantics == excludeFromSemantics
+        && typedOther.decoration == decoration
+        && typedOther.textStyle == textStyle
+        && typedOther.waitDuration == waitDuration
+        && typedOther.showDuration == showDuration;
+  }
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(DoubleProperty('height', height, defaultValue: null));
+    properties.add(DiagnosticsProperty<EdgeInsetsGeometry>('padding', padding, defaultValue: null));
+    properties.add(DoubleProperty('vertical offset', verticalOffset, defaultValue: null));
+    properties.add(FlagProperty('position', value: preferBelow, ifTrue: 'below', ifFalse: 'above', showName: true, defaultValue: null));
+    properties.add(FlagProperty('semantics', value: excludeFromSemantics, ifTrue: 'excluded', showName: true, defaultValue: null));
+    properties.add(DiagnosticsProperty<Duration>('wait duration', waitDuration, defaultValue: null));
+    properties.add(DiagnosticsProperty<Duration>('show duration', showDuration, defaultValue: null));
+  }
+}
+
+/// An inherited widget that defines the configuration for
+/// [Tooltip]s in this widget's subtree.
+///
+/// Values specified here are used for [Tooltip] properties that are not
+/// given an explicit non-null value.
+class TooltipTheme extends InheritedWidget {
+  /// Creates a tooltip theme that controls the configurations for
+  /// [Tooltip].
+  TooltipTheme({
+    Key key,
+    double height,
+    EdgeInsetsGeometry padding,
+    double verticalOffset,
+    bool preferBelow,
+    bool excludeFromSemantics,
+    Decoration decoration,
+    TextStyle textStyle,
+    Duration waitDuration,
+    Duration showDuration,
+    Widget child,
+  }) : data = TooltipThemeData(
+         height: height,
+         padding: padding,
+         verticalOffset: verticalOffset,
+         preferBelow: preferBelow,
+         excludeFromSemantics: excludeFromSemantics,
+         decoration: decoration,
+         textStyle: textStyle,
+         waitDuration: waitDuration,
+         showDuration: showDuration,
+       ),
+       super(key: key, child: child);
+
+  /// Specifies the properties for descendant [Tooltip] widgets.
+  final TooltipThemeData data;
+
+  /// The closest instance of this class that encloses the given context.
```

Comment:
This is the conventional boilerplate for an inherited widget; it's not correct for this theme. This `of` method returns the closest TooltipTheme ancestor's [data] value, and if there is no TooltipTheme ancestor, then it returns the current [Theme]'s tooltipTheme. Apps can assume that the returned value is not null.